### PR TITLE
Fix ncclDevFuncId for AllReduceWithBias

### PR DESCRIPTION
**Pull request by nileshnegi**
## Details

**Work item:**
Internal

**What were the changes?**  
- Update device codegen and host_table entry fields to incorporate bias/accumulator field.
- Removed mentions of `ncclFuncAllReduceWithBias` as we are re-purposing `ncclFuncAllReduce` for default vs. fused mode, based on accumulator field.
  - `ncclFuncAllReduceWithBias` still exists in api_trace symbols.

**Why were the changes made?**  
AllReduceWithBias was failing for `bfloat16` due to `ncclDevFuncId` lookup errors.

**Additional Documentation:**  
AllReduceWithBias will fail rccl-tests validation for fp8/int datatypes. ~Validation logic needs to be revisited~. To be fixed with https://github.com/ROCm/rccl-tests/pull/155

## Approval Checklist
___Do not approve until these items are satisfied.___
- [ ] Verify the CHANGELOG has been updated, if
  - there are any NCCL API version changes,
  - any changes impact library users, and/or
  - any changes impact any other ROCm library.

### CODE REVIEW

**Review by mberenjk — APPROVED** [review]

File: src/enqueue.cc [hunk 1]
```diff
@@ -501,13 +501,9 @@ ncclResult_t ncclPrepareTasks(struct ncclComm* comm, bool* algoNeedConnect, bool
 
       NCCLCHECK(getAlgoInfo(comm, &agg, collNetSupport, nvlsSupport, nTasksPerChannel, simInfo));
       if(agg.func==ncclFuncAllReduce && agg.acc != nullptr)
-      {
-        agg.devFuncId = ncclDevFuncId(ncclFuncAllReduceWithBias, agg.opDev.op, agg.datatype, agg.algorithm, agg.protocol, agg.pipeline);
-      }
+        agg.devFuncId = ncclDevFuncId(agg.func, agg.opDev.op, agg.datatype, agg.algorithm, agg.protocol, 1, agg.pipeline);
```
Comment (alex-breslow-amd) [inline src/enqueue.cc:504]: For readability consider defining constants like accEnabled=1 and accDisabled=0 to improve readability rather than just 1/0 followed by agg.pipeline

File: src/include/device.h [hunk 2]
```diff
@@ -159,7 +159,8 @@ static_assert(NCCL_LL_CLEAN_MASK % NCCL_STEPS == 0, "Invalid NCCL_LL_CLEAN_MASK
 #define RCCL_PROTO_SHIFT 8
 #define RCCL_REDOP_SHIFT 12
 #define RCCL_DTYPE_SHIFT 16
-#define RCCL_PIPELINE_SHIFT 20
+#define RCCL_ACC_SHIFT 20
```
Comment (alex-breslow-amd) [inline src/include/device.h:162]: I mentioned this when this code was written, but this code (generate.py + this code in device.h) needs at least some end-to-end tests to verify it works as intended.  It's too easy to bork this code path without hitting all of the corner cases.  I'll create a jira for this.

**Review by alex-breslow-amd — APPROVED** [review]
LGTM
